Choose Your Own Emoji.md

```diff
@@ -98,13 +98,32 @@ Html.Events.onClick (Update.SetSelectedKey emoji)
 
 When this element is clicked, a `Msg` of `Update.SetSelectedKey String` will be triggered, which will eventually be consumed by our application.
 
-This is a good time to talk about the strange value of the `Msg` union type that seems to have the type `String` attached to the end of it. We've seen this pattern once before — in the `SetCurrentText String` value of `Update.Msg`.
+Like `SetCurrentText String`, `SetSelectedKey String` is a **tagged value**. In effect, we are saying that a `SetCurrentText` and `SetSelectedKey` are only valid values of `Msg` union type if they are accompanies by a string.
 
-`SetCurrentText String` and `SetSelectedKey String` are **parameterized values**. In effect, we are saying that a `SetCurrentText` must always be accompanied by a string (which will become the model's new `currentText`), and that `SetSelectedKey` must always be accompanied by an emoji.
+Tagged values are helpful for modeling systems with uneven information requirements. For instance, if we were modeling a coffee shop, we might structure our data like this:
 
-By themselves, `SetCurrentText` and `SetSelectedKey` are not valid values for `Msg` — they require a parameter to function.
 
-So let's wire up this new `Msg` value to `Update.update`! In order:
+```elm
+type BrewingStrategy = PourOver | FrenchPress
+
+type Milk = Cow | Soy | Almond
+
+type alias NumberOfEspressoShots = Integer
+
+type DrinkOrder
+  = BlackCoffee BrewingStrategy
+  | Latte NumberOfEspressoShots Milk
+  | HotChocolate Milk
+  | ColdBrew
+```
+
+Apparently, there's a lot of complexity bundled up in our drink orders! In some cases, we need to know what type of milk the customer wants, in other cases, we need to know information specific to only one type of drink. The above data model accounts for that complexity by using tagged values — a `HotChocolate` is only a valid `DrinkOrder` if it is accompanied by a `Milk` selection, and a `BlackCoffee` relies on another union type to determine the customer's preferred brewing strategy.
```

Comment:
s/type of milk/kind of milk/

(just to keep it clear that all three kinds of milk are the same type)